### PR TITLE
multi_logical_optimizer: replace ListCopyDeep with copyObject

### PR DESCRIPTION
**Pull request by serprex**
Also stack allocate WorkerAggregateWalkerContext

`copyObject` does a deep copy: https://github.com/postgres/postgres/blob/a029a0641cce002acbb5e8ac6ec5533e599a9aea/src/backend/nodes/copyfuncs.c#L5137

### CODE REVIEW

File: src/backend/distributed/planner/multi_logical_optimizer.c [hunk 1]
```diff
@@ -2408,19 +2388,18 @@ ProcessHavingClauseForWorkerQuery(Node *originalHavingQual,
...
-		WorkerAggregateWalkerContext *workerAggContext = palloc0(
-			sizeof(WorkerAggregateWalkerContext));
+		WorkerAggregateWalkerContext workerAggContext = { 0 };
 
-		workerAggContext->extendedOpNodeProperties = extendedOpNodeProperties;
-		workerAggContext->expressionList = NIL;
-		workerAggContext->createGroupByClause = false;
+		workerAggContext.extendedOpNodeProperties = extendedOpNodeProperties,
+		workerAggContext.expressionList = NIL,
+		workerAggContext.createGroupByClause = false,
```
Comment (JelteF) [inline src/backend/distributed/planner/multi_logical_optimizer.c]: These assignments seem a bit useless
```suggestion
```

File: src/backend/distributed/planner/multi_logical_optimizer.c [hunk 2]
```diff
@@ -2408,19 +2388,18 @@ ProcessHavingClauseForWorkerQuery(Node *originalHavingQual,
...
 		 * then we need to combine the aggregates in the HAVING across shards.
 		 */
-		WorkerAggregateWalkerContext *workerAggContext = palloc0(
-			sizeof(WorkerAggregateWalkerContext));
+		WorkerAggregateWalkerContext workerAggContext = { 0 };
 
-		workerAggContext->extendedOpNodeProperties = extendedOpNodeProperties;
-		workerAggContext->expressionList = NIL;
-		workerAggContext->createGroupByClause = false;
+		workerAggContext.extendedOpNodeProperties = extendedOpNodeProperties,
```
Comment (JelteF) [inline src/backend/distributed/planner/multi_logical_optimizer.c]: Why does this line end in a comma? Is that on purpose?

File: src/backend/distributed/planner/multi_logical_optimizer.c [hunk 3]
```diff
@@ -2408,19 +2388,18 @@ ProcessHavingClauseForWorkerQuery(Node *originalHavingQual,
 		 * If the GROUP BY or PARTITION BY is not on the distribution column
 		 * then we need to combine the aggregates in the HAVING across shards.
 		 */
-		WorkerAggregateWalkerContext *workerAggContext = palloc0(
-			sizeof(WorkerAggregateWalkerContext));
+		WorkerAggregateWalkerContext workerAggContext = { 0 };
```
Comment (JelteF) [inline src/backend/distributed/planner/multi_logical_optimizer.c]: Maybe do this instead:
```c
WorkerAggregateWalkerContext workerAggContext = {
    .extendedOpNodeProperties = extendedOpNodeProperties
};

File: src/backend/distributed/planner/multi_logical_optimizer.c (hunk 2 above)
Comment (serprex) [reply src/backend/distributed/planner/multi_logical_optimizer.c]: Leftover from moving away from struct initializer syntax

File: src/backend/distributed/planner/multi_logical_optimizer.c (hunk 3 above)
Comment (serprex) [reply src/backend/distributed/planner/multi_logical_optimizer.c]: I'm confused now. I'd done that previously, but it's been replaced with `{ 0 }` & I was seeing you were putting memset calls to zero structs beforehand

Comment (JelteF) [reply src/backend/distributed/planner/multi_logical_optimizer.c]: Those were some special cases, that were flagged by static analysis. It's not necessary everywhere. The reason that they were flagged was that the the struct would later be returned by the function. This has the possible security issue that uninitiliazed stack memory leaves the current function, possibly ending up in some output. 

Here that the struct is not returned, so it's not necessary and we can use nice initializer syntax instead.

**Pull request merged**